docs/DESKTOP_ENVIRONMENTS.md

```diff
@@ -0,0 +1,118 @@
+
+## Pre-defined systemd units
+
+[`systemd.special(7)`](https://www.freedesktop.org/software/systemd/man/systemd.special.html)
+defines the `graphical-session.target` and `graphical-session-pre.target` to
+allow cross-desktop integration. Furthermore, systemd defines the three base
+slices `background`, `apps` and `session`. All units should be placed into
+one of these slices depending on their purposes:
+
+ * `session.slice`: Contains only processes essential to run the users graphical session
+ * `apps.slice`: Contains all normal applications that the user is running
+ * `background.slice`: Useful for low-priority background tasks
+
+The purpose of this grouping is to assign different priorities to the
+applications. This could e.g. mean reserving memory to session processes,
+preferentially killing background tasks in out-of-memory situations
+or assinging different memory/CPU/IO priorities to ensure that the session
+runs smoothly under load.
+
+TODO: Will there be a default to place units into e.g. `apps.slice` by default
+rather than the root slice?
+
+## XDG standardization for applications
+
+To ensure cross-desktop compatibility and encourage sharing of good practices,
+desktop environments should adhere to the following conventions:
+
+ * Application units should follow the scheme `apps-launcher-ApplicationID-PID.service`,
+   e.g. `apps-gnome-org.gnome.Evince-12345.service`,
+   `apps-flatpak-org.telegram.desktop-12345.service` or `apps-KDE-org.kde.okular-12345.service`.
+ * Use of `.scope` units is discouraged.
+ * Applications must be placed into the appropriate slice. It is suggested to use
+   the same naming scheme, and use the `apps-launcher-ApplicationID.slice`
+   (the PID may be included optionally, simplifying the unit creation as it
+   can be done by setting `Slice=%N.slice` in a drop-in configuration).
+ * If no application ID is available, the launcher should generate a reasonable
+   name when possible (e.g. using `basename(argv[0])`).
+
+TODO: Define the name of slices that should be used. This could be `apps-launcher-ApplicationID.slice`.
+
+This has the following advantages:
+ * Using the `apps-launcher-` prefix means that the unit defaults can be adjusted
+   using desktop environment specific drop-in files.
+ * The application ID can be retrieved by stripping the prefix and postfix.
+   This in turn should map to the corresponding `.desktop` file when available
+
+## XDG autostart integration
+
+To allow XDG autostart integration, systemd will ship a cross-desktop generator
+to create appropriate units for the autostart directory. Desktop Environments
+will be able to make use of this simply by starting the appropriate XDG related
+targets. The names and ordering rules for these targets are to be defined.
+
+This generator will likely only support a subset of features. In particular,
+the GNOME specific feature to bind a service to a settings variable will not
+be supported.
+
```

Comment:
We discussed a clever idea that I don't want forgotten.

If there's an X-Only-Show in it'll put symlinks in the kde.target or gnome.target or whatever as relevant.

Otherwise they'll go in the default.target   

Then we as Plasma include kde and default, and get the righ things.

I also want us to define how we resolve duplicates of things with a dedicated .service file

Reply:
Yes, my idea for this is that desktop environments should start a `xdg-desktop-X.target` pseudo targets for each `$XDG_CURRENT_DESKTOP` entry.